### PR TITLE
chore: update sync to pg workflow to use github pat for accessing the secrets

### PR DESCRIPTION
**Pull request by AnaghHegde**
## Description
In order to modify a workflow, a GitHub App, such as the one used for issuing tokens for GitHub Actions, requires the workflow scope. This is so that GitHub Apps you've added to your repository can't access the secrets in your repository without your permission. The token issued for GitHub Actions doesn't have this permission by default.
This PR adds pat token to the push command. This is to avoid failures when the secrets are modified in the repositories. 
## Automation

/ok-to-test tags=""

### :mag: Cypress test results

> [!WARNING]
> Tests have not run on the HEAD b09d3542da73c527a1a52c93abb3b06a595c3bd3 yet
> <hr>Wed, 13 Nov 2024 06:09:54 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **Chores**
	- Updated the push command for merging the `release` branch into the `pg` branch to use a personal access token for enhanced security.
	- Modified error message capture to align with the new push command for better error tracking.

### CODE REVIEW

File: .github/workflows/sync-release-to-pg.yml [hunk 1]
```diff
@@ -50,7 +50,7 @@ jobs:
       - name: Push changes
         if: env.MERGE_CONFLICT == 'false'
         run: |
-          git push origin pg
+          git push https://${{ secrets.PAT_GITHUB }}@github.com/${{ github.repository }} HEAD:pg
```
Comment (abhvsn) [inline .github/workflows/sync-release-to-pg.yml:53]: What is `PAT_GITHUB`? I can't see this in secrets may be you are looking at EE repo where this is present.

Comment (abhvsn) [reply .github/workflows/sync-release-to-pg.yml:53]: Also we may want to add the github.actor in the url:
```
git remote set-url origin 'https://${{ github.actor }}:${{ secrets.PAT_GITHUB }}@github.com/${{ github.repository }}.git'
```

Comment (AnaghHegde) [reply .github/workflows/sync-release-to-pg.yml:53]: You are right, it must be present in EE only. Do you have access to create this?

File: .github/workflows/sync-release-to-pg.yml [hunk 2]
```diff
@@ -51,13 +51,13 @@
         if: env.MERGE_CONFLICT == 'false'
         run: |
           set -e
-          git push origin pg || echo "PUSH_FAILURE=true" >> $GITHUB_ENV
+          git push https://${{ secrets.PAT_GITHUB }}@github.com/${{ github.repository }} HEAD:pg || echo "PUSH_FAILURE=true" >> $GITHUB_ENV
```
[the remark this answers is not in the document]
Comment (abhvsn) [reply .github/workflows/sync-release-to-pg.yml:54]: @AnaghHegde have you checked this comment.

**Review by abhvsn — APPROVED** [review]
Check the bot review else LGTM.

**Pull request merged**